File: app/src/main/java/org/illegaller/ratabb/hishoot2i/view/common/Mvp.java
package org.illegaller.ratabb.hishoot2i.view.common;

import android.content.Context;

public class Mvp {
  public interface View {
    Context getContext();
  }

  interface Presenter<V extends View> {
    void attachView(V mvpView);

    void detachView();
  }
}

File: app/src/main/java/org/illegaller/ratabb/hishoot2i/ui/common/Mvp.kt
package org.illegaller.ratabb.hishoot2i.ui.common

import kotlinx.coroutines.CoroutineScope

interface Mvp {
    interface View {
        fun onError(e: Throwable)
    }

    interface Presenter<V : View> : CoroutineScope {
        fun attachView(view: V)
        fun detachView()
    }
}
